scripts/album.js

```diff
@@ -51,7 +51,58 @@ var createSongRow = function(songNumber, songName, songLength) {
   + '</tr>'
   ;
 
-  return $(template);
+  var $row = $(template);
+
+  var clickHandler = function() {
+    var $songItem = $(this).find('.song-item-number');
+    var $songItemNumber = $songItem.attr('data-song-number');
+
+    // if there isn't a song playing, set the clicked song item number to pause
+    if (currentlyPlayingSong === null) {
+      $songItem.html(pauseButtonTemplate);
+      currentlyPlayingSong = $songItemNumber
+
+    // if the song that's playing is the song we've clicked, change to play button
+    } else if (currentlyPlayingSong === $songItemNumber) {
+      $songItem.html(playButtonTemplate);
+      currentlyPlayingSong = null;
+
+    // if the song that's playing is not the song we've clicked, change the clicked song's number
+    // to pause button and change the previously playing song's pause button to number
+    } else if (currentlyPlayingSong !== $songItemNumber) {
+      var $currentlyPlayingSongElement = $('[data-song-number="' + currentlyPlayingSong + '"]');
+      $currentlyPlayingSongElement.html($currentlyPlayingSongElement.attr('data-song-number'));
+      $songItem.html(pauseButtonTemplate);
+      currentlyPlayingSong = $songItemNumber;
+    }
+  };
+
+  var onHover = function(event) {
+    // we don't need to check that the parent element matches ".album-view-song-item"
+    //   because we're calling onHover on $row, which represents "this" below
+    //   and $row will always have a the same structure (see createSongRow)
+    var $songItem = $(this).find(".song-item-number");
+    var songItemNumber = $songItem.attr('data-song-number');
+
+    if (songItemNumber !== currentlyPlayingSong) {
+      $songItem.html(playButtonTemplate);
+    }
+  };
+
+  var offHover = function(event) {
+    var $songItem = $(this).find(".song-item-number");
+    var songItemNumber = $songItem.attr('data-song-number');
+
+    if (songItemNumber !== currentlyPlayingSong) {
+      $songItem.html(songItemNumber);
+    }
+  };
+
+  // This is how it's setup in the checkpoint --
+  //   $row.find('.song-item-number').click(clickHandler);
+  $row.click(clickHandler);
```

Comment:
So this different and at least partially responsible for the problem. I think it's preferable to place the event listener on the `$row` though, no? If it's only on `.song-item-number` elements, it won't fire when I click on the title, or duration as it did in the previous implementation.